python/air/dialects/_air_ops_ext.py

```diff
@@ -48,7 +71,7 @@ def __init__(
             launch_operands=operands,
             sym_name=name,
         )
-        operand_types = [s.type for s in sizes] * 2 + [o.type for o in operands]
+        operand_types = [s.type for s in sizes] * 2 + get_region_operand_types(operands)
```

Comment:
I think considering not just Values but also Operations makes sense in the segment and herd since one could feasibly create the operation in the outer region. However, the launch is the outermost region in which an operation could occur (to my knowledge) so I suspect that we should only allow Values here.

Does this reasoning make sense? if so, I'll remove the change to the `Launch()` class.

Reply:
@jgmelber Do you have an opinion?

Reply:
I think in the future there could be other operations before or after launch.

Reply:
Okay, I will keep my changes then!

Reply:
Maybe I'm missing some subtlety about the python bindings but isn't the result of an Operation always one or more Values? Can you share a code snippet where this comes up? This isn't an issue in ordinary MLIR code or using C++ APIs.

Reply:
Here is the context: https://github.com/Xilinx/mlir-air/issues/666